week-2/Homework/extra/2-array-methods.js

```diff
@@ -5,31 +5,72 @@ Following are set of small functions that you need to create using loops and arr
 //1)Find the Smallest Number in an Array
 //Create a function that takes an array of numbers and return the smallest number in it
 
-function findSmallestNum(arr) {}
+function findSmallestNum(arr) {
+    let min = arr[0];

```

Comment:
Overall is great, however, we as programmers sometimes we need to think about edge cases or boundary cases in order to make this function rock solid within a sensible limit. We haven't stipulated in our function contract how many elements the array can have so it can be from zero to "infinite".

What if it is empty? What'd occur in that case?

Reply:
I can think of a solution for the Zero case, but for the infinte, my solution would be to wrap the whole body of the function inside of a `if (arr.length >0 ) { body of the function }`

Reply:
It is a good approach! If array is empty, the function would return `undefined` (I believe so)

There are other technics as well, but it is not out of the scope of the subject.